### PR TITLE
Fix match detail

### PR DESCRIPTION
**Pull request by cp-mayank**
- Match detail
      - info tab first
      - commantary tab default
      - keep scorecard expanded
      - update squad tab player ui
      - show last overView
      - change commantary ball by ball ui according highlights           
      
[match_detail.webm](https://github.com/canopas/khelo/assets/140152611/cb6789c8-eff4-483c-8d05-07e2b181ff6c)
     

## Summary by CodeRabbit

- **New Features**
  - Updated the `selectedTab` default value in the match detail tab view to focus on `matchInfo`.
  - Added a new parameter `isSecondTeam` in the squad view for better team differentiation.

- **Improvements**
  - Enhanced commentary view layout with additional padding and dividers.
  - Improved scorecard handling by refining power play overs storage.
  - Adjusted player profile view's layout and alignment based on team context.
  
- **Bug Fixes**
  - Fixed layout inconsistencies in the ball summary display.

### CODE REVIEW

**Comment by cp-sidhdhi-p** [comment]
@cp-mayank  I think we should show ball score as well in commentary.

File: khelo/lib/ui/flow/matches/match_detail/components/match_detail_scorecard_view.dart [hunk 1]
```diff
@@ -78,20 +87,12 @@ class MatchDetailScorecardView extends ConsumerWidget {
...
-                if (state.match?.power_play_overs3.isNotEmpty ?? false)
-                  state.match!.power_play_overs3,
-              ];
-
               return _teamTitleView(context,
                   teamName: _getTeamNameByTeamId(state, overs?.team_id ?? ""),
                   over: overs ?? const OverSummary(),
-                  initiallyExpanded: state.expandedTeamScorecard
-                      .contains(overs?.team_id ?? ""),
+                  initiallyExpanded: groupOversByInnings.length == 1 ||
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/matches/match_detail/components/match_detail_scorecard_view.dart:93]: - if I close the card and change tab then it should remain the closed but it 's opened, which is not correct behaviour.
- also the icon is not animating for open and closed states, can you fix it as well.
iconAnimationIssue

**Comment by cp-sidhdhi-p** [comment]
@cp-mayank please hide unwanted dividers like below:
Simulator Screenshot - iPhone 15 - 2024-07-08 at 15 00 13

File: khelo/lib/ui/flow/matches/match_detail/match_detail_tab_view_model.dart [hunk 2]
```diff
@@ -81,7 +81,7 @@ class MatchDetailTabViewNotifier extends StateNotifier<MatchDetailTabState> {
             .firstOrNull;
 
         onScorecardExpansionChange(
-          match.matchResult?.teamId ?? match.teams.firstOrNull?.team.id ?? "",
+          match.matchResult?.teamId ?? match.teams.last.team.id ?? "",
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/matches/match_detail/match_detail_tab_view_model.dart:84]: what this change is supposed to do?

**Comment by cp-sidhdhi-p** [comment]
can you fix this line break when user name is long.
linebreaks

File: khelo/lib/ui/flow/matches/match_detail/match_detail_tab_view_model.dart (hunk 2 above)
Comment (cp-mayank) [reply khelo/lib/ui/flow/matches/match_detail/match_detail_tab_view_model.dart:84]: If there is only one team in the scorecard, then pass the team ID for default expansion.

Comment (cp-sidhdhi-p) [reply khelo/lib/ui/flow/matches/match_detail/match_detail_tab_view_model.dart:84]: ` match.teams.last.team.id ?? ""`
match.teams always have 2 teams, you can use currentplaying teamId for our usecase

**Review by cp-sidhdhi-p — APPROVED** [review]

**Pull request merged**